plugin-packs/postcss-bundler/src/postcss-import/lib/apply-styles.ts

```diff
@@ -1,16 +1,21 @@
 import type { Document, Root } from 'postcss';
-import { Statement, isCharsetStatement, isImportStatement } from './statement';
+import { Stylesheet, isImportStatement, isNodesStatement } from './statement';
 
-export function applyStyles(bundle: Array<Statement>, styles: Root | Document) {
+export function applyStyles(stylesheet: Stylesheet, styles: Root | Document) {
 	styles.nodes = [];
 
+	if (stylesheet.charset) {
+		stylesheet.charset.parent = undefined;
+		styles.append(stylesheet.charset);
+	}
```

Comment:
`@charset` must always be first.

It doesn't make sense to add it in a loop where theoretically it could be inserted later.